dags/sql_examples/sql_check_redshift_etl.py

```diff
@@ -0,0 +1,244 @@
+from airflow.utils.dates import datetime
+from airflow.operators.sql import (
+    SQLCheckOperator,
+    SQLValueCheckOperator,
+    SQLIntervalCheckOperator,
+    SQLThresholdCheckOperator
+)
+from airflow.utils.task_group import TaskGroup
+from airflow.models import Variable
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+from airflow.providers.amazon.aws.transfers.s3_to_redshift import S3ToRedshiftOperator
+from airflow.providers.postgres.operators.postgres import PostgresOperator
+from airflow.decorators import task
+
+import pandas as pd
+
+
+DATES = ["2019-01", "2019-02"]
+TASK_DICT = {}
+
+with DAG("sql_data_quality",
+         start_date=datetime(2021, 7, 7),
+         description="A sample Airflow DAG to perform data quality checks using SQL Operators.",
+         schedule_interval=None,
+         default_args={"conn_id": "redshift_default"},
+         catchup=False) as dag:
+    """
+    ### SQL Check Operators Data Quality ETL Example
+
+    Before running the DAG, set the following in an Airflow or Environment Variable:
+    - key: aws_configs
+    - value: { "s3_bucket": [bucket_name], "s3_key_prefix": [key_prefix], "redshift_table": [table_name]}
+    Fully replacing [bucket_name], [key_prefix], and [table_name].
+
+    See the README for information on how to set up your Redshift connection.
+    This DAG can be used with other databases as long as the Redshift (and possibly
+    transfer operators) are changed.
+    """
+
+    begin = DummyOperator(task_id="begin")
+    end = DummyOperator(task_id="end")
+    converge_1 = DummyOperator(task_id="converge_1")
+    converge_2 = DummyOperator(task_id="converge_2")
+
+    @task
+    def add_upload_date(file_path, upload_date):
+        """
+        #### Transform Task
+        In general, it is not recommended to perform transform operations in
+        Airflow Tasks, as Airflow is designed to be an orchestrator, not a
+        computation engine. However, the transform is done here as it is a
+        relatively small operation, simply adding an upload_date column to the
+        dataframe for use in the SQL data quality checks later. Doing the
+        transform here also makes this example more easily extensible to the
+        use of other backend datastores.
+        """
+        trip_dict = pd.read_csv(
+            file_path,
+            header=0,
+            parse_dates=["pickup_datetime"],
+            infer_datetime_format=True
+        )
+        trip_dict["upload_date"] = upload_date
+        trip_dict.to_csv(
+            file_path,
+            header=True,
+            index=False
+        )
+
+    add_upload_date_1 = add_upload_date(
+        "/usr/local/airflow/include/data/yellow_tripdata_sample_2019-01.csv",
+        "{{ ds }}"
+    )
+    add_upload_date_2 = add_upload_date(
+        "/usr/local/airflow/include/data/yellow_tripdata_sample_2019-02.csv",
+        "{{ yesterday_ds }}"
+    )
+
+    @task
+    def upload_to_s3(file_path):
```

Comment:
Would using the [LocalFilesystemToS3Operator](https://registry.astronomer.io/providers/amazon/modules/localfilesystemtos3operator) be useful here?

Reply:
Yes haha, and I wrote that for this particular use case. This DAG was originally just to help me test the operators, then I decided to actually use it, and forgot to implement that after copying all this from the Redshift DAG (which I also need to update with the LocalFileSystemtoS3Operator).